### PR TITLE
Implemented Toast message on Back pressed in Deck Picker activity

### PR DESCRIPTION
**Pull request by iamvs-2002**
## Purpose / Description
_Describe the problem or feature and motivation_
Implemented Toast message "Press the button again to exit" on Back pressed in Deck Picker activity
## Fixes
Fixes _Link to the issues._
#8226 
## Approach
_How does this change address the problem?_

## How Has This Been Tested?
I have tested this on Google Pixel XL Api 30 android device.
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by iamvs-2002** [comment]
https://user-images.githubusercontent.com/54304786/113481738-e9a91e00-94b8-11eb-9aca-402998373be0.mp4

That's how the output looks.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 1]
```diff
@@ -1023,18 +1023,26 @@ private void automaticSync() {
         }
     }
 
+    private long pressedTime;
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1026]: > * If the user presses 'Back' once on the DeckPicker, a Toast should be displayed stating Press back again to close
> * There is no functional difference between how "back" works now, and how pressing it a second time (after this change) will work

There wasn't anything about a time

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 2]
```diff
@@ -1023,18 +1023,26 @@ private void automaticSync() {
...
     @Override
     public void onBackPressed() {
         if (isDrawerOpen()) {
             super.onBackPressed();
         } else {
             Timber.i("Back key pressed");
             if (mFloatingActionMenu.isFABOpen()) {
+                Toast.makeText(this, "Press back again to exit.", Toast.LENGTH_SHORT).show();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1034]: Could you extract these to resources

https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#adding-translations

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Thanks!

**Comment by david-allison-1** [comment]
```
  /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1037: Error: Use the collection's getTime() method instead of System.currentTimeMillis() [DirectSystemCurrentTimeMillisUsage]
                  if (pressedTime + 2000 > System.currentTimeMillis()){
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1045: Error: Use the collection's getTime() method instead of System.currentTimeMillis() [DirectSystemCurrentTimeMillisUsage]
              pressedTime = System.currentTimeMillis();
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java (hunk 1 above)
Comment (iamvs-2002) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1026]: The time here refers to the duration in which the user should press the back button, a second time in order to exit. I have set this time = 2000 milliseconds.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java (hunk 2 above)
Comment (iamvs-2002) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1034]: Sure

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1026]: That isn't what's in the bullet points above

Comment (iamvs-2002) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1026]: I thought it was good, so I implemented it, let me do it as you say.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 3]
```diff
@@ -1023,18 +1023,26 @@ private void automaticSync() {
         }
     }
 
+    public int count = 1;
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1026]: use a boolean

File: AnkiDroid/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -6,4 +6,5 @@
     <!-- https://developers.google.com/analytics/devguides/collection/android/v4/limits-quotas -->
     <integer name="ga_sampleFrequency">10</integer>
     <bool name="ga_anonymizeIp">true</bool>
+    <string name="toastOnBackPressed" translatable="false">Press back again to exit.</string>
```
Comment (david-allison-1) [inline AnkiDroid/src/main/res/values/strings.xml:9]: > Keys names should be lowercase using underscores

This is in the wrong file

>  1. Determine the correct resource XML file to edit ([Guide](https://github.com/ankidroid/Anki-Android/wiki/Translating-AnkiDroid#logic-of-the-separation-in-different-files)) and expand it in Android Studio
> 2. Add the string to the first file in the list (`/values`) and no other file.


https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#adding-translations

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 5]
```diff
@@ -1023,18 +1023,26 @@ private void automaticSync() {
...
     @Override
     public void onBackPressed() {
         if (isDrawerOpen()) {
             super.onBackPressed();
         } else {
             Timber.i("Back key pressed");
             if (mFloatingActionMenu.isFABOpen()) {
+                Toast.makeText(this, getApplication().getString(R.string.toastOnBackPressed), Toast.LENGTH_SHORT).show();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1034]: Use `UIUtils.showThemedToast`

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 6]
```diff
@@ -1023,18 +1023,26 @@ private void automaticSync() {
...
                 mFloatingActionMenu.closeFloatingActionMenu();
             } else {
-                automaticSync();
-                finishWithAnimation();
+                if (count!=1){
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1037]:
```suggestion
                if (count != 1) {
```

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 7]
```diff
@@ -1023,26 +1023,26 @@ private void automaticSync() {
         }
     }
 
-    public int count = 1;
+    public boolean count = false;
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1026]: Could you move the variable to the top of the file

Could you give the variable a name in line with our naming conventions: https://github.com/ankidroid/Anki-Android/wiki/Code-style#non-public-non-static-field-names-should-start-with-m

`mBackButtonPressedToExit`

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 8]
```diff
@@ -1023,26 +1023,26 @@ private void automaticSync() {
...
     @Override
     public void onBackPressed() {
         if (isDrawerOpen()) {
             super.onBackPressed();
         } else {
             Timber.i("Back key pressed");
             if (mFloatingActionMenu.isFABOpen()) {
-                Toast.makeText(this, getApplication().getString(R.string.toastOnBackPressed), Toast.LENGTH_SHORT).show();
+                UIUtils.showThemedToast(this, getApplication().getString(R.string.back_pressed_once),true);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1034]:
```suggestion
                UIUtils.showThemedToast(this, getApplication().getString(R.string.back_pressed_once), true);
```

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 9]
```diff
@@ -1023,26 +1023,26 @@ private void automaticSync() {
...
                 mFloatingActionMenu.closeFloatingActionMenu();
             } else {
-                if (count!=1){
+                if (count){
                     automaticSync();
                     finishWithAnimation();
                 }
                 else {
-                    Toast.makeText(this, getApplication().getString(R.string.toastOnBackPressed), Toast.LENGTH_SHORT).show();
+                    UIUtils.showThemedToast(this, getApplication().getString(R.string.back_pressed_once),true);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1042]:
```suggestion
                    UIUtils.showThemedToast(this, getApplication().getString(R.string.back_pressed_once), true);
```

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
This appears to be missing a resource, so it's unlikely it'll compile

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java (hunk 7 above)
Comment (iamvs-2002) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1026]: Alright, I am on it

**Comment by iamvs-2002** [comment]
> ~This appears to be missing a resource, so it's unlikely it'll compile~ EDIT: Incorrect
> 
> This contains changes from another PR

Oh god...I think I again combined two branches.

**Comment by david-allison-1** [comment]
Let me know when things are sorted

**Comment by iamvs-2002** [comment]
> Let me know when things are sorted

I tried to rebase on the remote, kindly check it out once.

**Comment by iamvs-2002** [comment]
> > Let me know when things are sorted
> 
> I tried to rebase on the remote, kindly check it out once.

If this has worked, it's fine, else it means I messed up the branches, and hence, I'll again do it.

**Comment by david-allison-1** [comment]
You can see the files and tell if it's worked: https://github.com/ankidroid/Anki-Android/pull/8453/files

**Comment by iamvs-2002** [comment]
> You can see the files and tell if it's worked: https://github.com/ankidroid/Anki-Android/pull/8453/files.
> 
> You'll likely want to `rebase -i` and `drop` commits.

I tried doing this, but I had already rebased it, due to which it doesn't change any more. I'll do it all again once the previous PR for some other issue is completely approved so that I'll be focussing on only one branch.

**Comment by david-allison-1** [comment]
> > You can see the files and tell if it's worked: https://github.com/ankidroid/Anki-Android/pull/8453/files.
> > You'll likely want to `rebase -i` and `drop` commits.
> 
> I tried doing this, but I had already rebased it, due to which it doesn't change any more. I'll do it all again once the previous PR for some other issue is completely approved so that I'll be focussing on only one branch.

At the worst case, you can `git reset`, and `git cherry-pick` your changes

**Comment by iamvs-2002** [comment]
> > > You can see the files and tell if it's worked: https://github.com/ankidroid/Anki-Android/pull/8453/files.
> > > You'll likely want to `rebase -i` and `drop` commits.
> > 
> > 
> > I tried doing this, but I had already rebased it, due to which it doesn't change any more. I'll do it all again once the previous PR for some other issue is completely approved so that I'll be focussing on only one branch.
> 
> At the worst case, you can `git reset`, and `git cherry-pick` your changes back

Ohh let me do that finally then.

**Comment by iamvs-2002** [comment]
I have finally resolved it. I'll close this pull request and create a new pull request, also Ill mention this pull request there.

**Comment by david-allison-1** [comment]
> I have finally resolved it. I'll close this pull request and create a new pull request, also Ill mention this pull request there.

You should just be able to force push to the branch that the PR is off

**Pull request closed without merging**